File: Data Structures & Algorithms/largest-unique-number/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    largestUniqueNumber(nums) {
        const m = {}

        for (let v of nums) {
            m[v] = (m[v] || 0) + 1
        }

        let max = -1
        for (let v of Object.keys(m)) {
            if (m[v] === 1) {
                max = Math.max(max, Number(v))
            }  
        }

        return max
    }
}

File: Data Structures & Algorithms/maximum-product-subarray/submission-0.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
      if not nums: 
        return 0
      
      res = cur_max = cur_min = nums[0]
      for num in nums[1:]:
        opt1 = num
        opt2 = num * cur_max
        opt3 = num * cur_min

        cur_max = max(opt1, opt2, opt3)
        cur_min = min(opt1, opt2, opt3)
        res = max(res, cur_max)


      return res